src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -204,6 +211,85 @@ public SpecificHaxeClassReference getUnderlyingClassReference(HaxeGenericResolve
     return types;
   }
 
+  public List<SpecificHaxeClassReference> getCastableToTypesList(SpecificHaxeClassReference sourceType) {
+    if (!isAbstract()) return Collections.emptyList();
+    List<HaxeMethodModel> methodsWithMetadata = getCastToMethods();
+
+    return  methodsWithMetadata.stream()
+      .filter(methodModel -> castMethodAcceptsSource(sourceType, methodModel))
+      .map(m -> SetSpecificsConstraints(m, getReturnType(m)))
+      .collect(toList());
+  }
+
+
+  private boolean castMethodAcceptsSource(SpecificHaxeClassReference reference, HaxeMethodModel methodModel) {
```

Comment:
You might consider using `@NotNull` and `@Nullable` annotations on the new methods.